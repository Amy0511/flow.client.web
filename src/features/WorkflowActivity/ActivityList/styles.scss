.c-activity-list {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 1.5rem;
}
.bmrg--c-loading-animation.s-activities-loading{
  height: 3rem;
  justify-content: center;
}
.s-activities-loading{
  .bmrg--b-loading-animation__text{
    display: none;
  }.bmrg--b-loading-animation__img{
    transform: scale(0.3);
  }
}
