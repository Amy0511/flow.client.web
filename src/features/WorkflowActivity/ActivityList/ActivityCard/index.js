import React, { Component } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import moment from "moment";
import getHumanizedDuration from "@boomerang/boomerang-utilities/lib/getHumanizedDuration";
import { ACTIVITY_STATUSES_TO_TEXT, ACTIVITY_STATUSES_TO_ICON } from "Constants/activityStatuses";
import imgs from "Assets/icons";
import "./styles.scss";

class ActivityCard extends Component {
  static propTypes = {
    activity: PropTypes.object.isRequired,
    match: PropTypes.object.isRequired
  };

  render() {
    const {
      description,
      creationDate,
      icon,
      duration = 0,
      id,
      initiatedByUserName,
      status,
      teamName,
      trigger,
      workflowId,
      workflowName
    } = this.props.activity;
    return (
      <Link
        className={`c-activities-card --${status}`}
        to={{
          pathname: `/activity/${workflowId}/execution/${id}`,
          state: { fromUrl: this.props.match.url, fromText: "Activity" }
        }}
      >
        <div className="c-activity-card-workflow">
          <div className="c-activity-card-section--left">
            <div className="c-activity-card__icon">
              <img className="b-activity-card__icon" src={imgs[icon ? icon : "docs"]} alt="Worklflow icon" />
            </div>
            <div className="c-activity-card-workflow-info">
              <label className="b-activity-card__name">{workflowName}</label>
              <label className="b-activity-card__description">{description}</label>
            </div>
          </div>
          <div className="c-activity-card-section--right">
            <ul className="c-activity-card-execution-info --first">
              <li className="b-activity-card__row">
                <label className="b-activity-card__label">Team</label>
                <label className="b-activity-card__data">{teamName}</label>
              </li>
              <li className="b-activity-card__row">
                <label className="b-activity-card__label">Trigger</label>
                <label className="b-activity-card__data">
                  {trigger ? trigger.slice(0, 1).toUpperCase() + trigger.slice(1) : "---"}
                </label>
              </li>
              <li className="b-activity-card__row">
                <label className="b-activity-card__label">Initiated by</label>
                <label className="b-activity-card__data">{initiatedByUserName || "---"}</label>
              </li>
            </ul>
            <ul className="c-activity-card-execution-info">
              <li className="b-activity-card__row">
                <label className="b-activity-card__label">Start Time</label>
                <label className="b-activity-card__data">{moment(creationDate).format("YYYY-MM-DD hh:mm A")}</label>
              </li>
              <li className="b-activity-card__row">
                <label className="b-activity-card__label">Duration</label>
                <label className="b-activity-card__data">
                  {duration ? getHumanizedDuration(parseInt(duration / 1000, 10)) : "---"}
                </label>
              </li>
              <li className="b-activity-card__row">
                <label className="b-activity-card__label"> Status</label>
                <label className="b-activity-card__data">
                  <img
                    className="b-activity-card__status-icon"
                    src={ACTIVITY_STATUSES_TO_ICON[status ? status : "notstarted"]}
                    alt={`Status ${status}`}
                  />
                  {ACTIVITY_STATUSES_TO_TEXT[status ? status : "notstarted"]}
                </label>
              </li>
            </ul>
          </div>
        </div>
      </Link>
    );
  }
}

export default ActivityCard;
