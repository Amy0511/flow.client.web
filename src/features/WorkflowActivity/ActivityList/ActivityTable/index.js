import React, { Component } from "react";
import PropTypes from "prop-types";
import moment from "moment";
import Table from "@boomerang/boomerang-components/lib/Table";
import getHumanizedDuration from "@boomerang/boomerang-utilities/lib/getHumanizedDuration";
import activityStatuses from "Constants/activityStatuses";
import { BASE_SERVICE_URL } from "Config/servicesConfig";
import imgs from "Assets/icons";
import statusImgs from "./img";
import "./styles.scss";

class ActivityTable extends Component {
  static propTypes = {
    activities: PropTypes.array.isRequired,
    pageSize: PropTypes.number.isRequired,
    pageNumber: PropTypes.number.isRequired,
    totalPageCount: PropTypes.number.isRequired,
    savePageSize: PropTypes.func.isRequired,
    fetchActivities: PropTypes.func.isRequired,
    searchQuery: PropTypes.string,
    sort: PropTypes.array,
    totalElements: PropTypes.number
  };

  state = {
    pageSize: this.props.pageSize || 10,
    sorted: this.props.sort[0] ? this.props.sort[0].property : "",
    direction: this.props.sort[0] ? this.props.sort[0].direction : ""
  };

  getStatusIcon = (status) => {
    switch(status){
      case activityStatuses.COMPLETED:
        return statusImgs["passed"];
      case activityStatuses.FAILED:
        return statusImgs["failed"];
      default:
        return statusImgs["pending"];
    }
  }
  getStatusText = (status) => {
    switch(status){
      case activityStatuses.COMPLETED:
        return "Success";
      case activityStatuses.FAILED:
        return "Failed";
      default:
        return "In Progress";
    }
  }

  onSortedChange = (newSorted, column) => {
    const direction = newSorted[0].desc ? "DESC" : "ASC";

    this.setState({ sorted: column.id, direction }, () => {
      this.onPageChange(this.props.pageNumber);
    });
  };

  onPageChange = page => {
    const { pageSize, sorted, direction } = this.state;
    const { searchQuery } = this.props;
    this.props.fetchActivities(
      `${BASE_SERVICE_URL}/activity?size=${pageSize}&page=${page}&sort=${sorted}&order=${direction}&query=${searchQuery}`
    );
  };

  onPageSizeChange = pageSize => {
    this.props.savePageSize(pageSize);
    this.setState({ pageSize }, () => {
      this.onPageChange(0);
    });
  };

  columns = [
    {
      Header: "Workflow",
      accessor: "workflowName",
      className: "c-activities-row__item-workflow",
      width: 380,
      Cell: row => {
        return (
          <div className="c-activity-card__workflow">
            <div className="c-activity-card__icon">
              <img className="b-activity-card__icon" src={imgs[row.original.icon]} alt="icon"/>
            </div>
            <div className="c-activity-card__workflow-info">
              <label className="b-activity-card__name">{row.original.workflowName}</label>
              <label className="b-activity-card__description">{row.original.description}</label>
            </div>
          </div>
        );
      }
    },
    {
      Header: "Creation Date",
      accessor: "creationDate",
      className: "c-activities-row__item-activity",
      width: 380,
      Cell: row => {
        return (
          <div className="c-activity-card__activity">
            <div className="c-activity-card__label">
              <label className="b-activity-card__label">Start Time</label>
              <label className="b-activity-card__data">{moment(row.original.creationDate).format("YYYY-MM-DD hh:mm A")}</label>
            </div>
            <div className="c-activity-card__label">
              <label className="b-activity-card__label">Duration</label>
              <label className="b-activity-card__data">{getHumanizedDuration(row.original.duration)}</label>
            </div>
            <div className="c-activity-card__label">
              <label className="b-activity-card__label"> Status</label>
              <label className="b-activity-card__data">
                <img className="b-activity-card__status-icon" src={this.getStatusIcon(row.original.status)} alt="status" />
                {this.getStatusText(row.original.status)}
              </label>
            </div>
          </div>
        );
      }
    }
  ];

  render() {
    const { sort } = this.props;
    return (
        <div className="c-activities-table">
          <Table
            manual
            className=""
            data={this.props.activities}
            columns={this.columns}
            showPagination={this.props.activities.length > 0 || this.state.pageSize > 0 ? true : false}
            defaultPageSize={this.state.pageSize}
            minRows={this.props.activities && this.props.activities.length ? this.props.activities.length : 5}
            page={this.props.pageNumber}
            pages={this.props.totalPageCount}
            onPageChange={this.onPageChange}
            onPageSizeChange={this.onPageSizeChange}
            onSortedChange={this.onSortedChange}
            noDataText="No audit records found"
            theme="bmrg-blue"
            getTrGroupProps={(state, rowInfo) => {
              return {className: `--${rowInfo.original.status}`}
            }}
            defaultSorted={
              sort
                ? sort.map(sort => ({
                    id: sort.property,
                    desc: sort.descending
                  }))
                : []
            }
            
          />
        </div>
    );
  }
}

export default ActivityTable;
