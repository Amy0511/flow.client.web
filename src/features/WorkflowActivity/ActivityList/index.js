import React, { Component } from "react";
import PropTypes from "prop-types";
import axios from "axios";
// import ActivityTable from "./ActivityTable";
import LoadingAnimation from "@boomerang/boomerang-components/lib/LoadingAnimation";
import InfiniteScroll from 'react-infinite-scroller';
import ActivityCard from "./ActivityCard";
import ScrollUp from "Components/ScrollUp";
import { BASE_SERVICE_URL } from "Config/servicesConfig";
import "./styles.scss";

class ActivityList extends Component {

  static propTypes = {
    activities: PropTypes.array.isRequired,
    history: PropTypes.object.isRequired,
    searchQuery: PropTypes.string
  };
  state = {
    hasMoreActivities: !this.props.activities.last,
    activitiesList: this.props.activities.records
  };

  loadMoreActivities = (page) => {
    let newActivities = [].concat(this.state.activitiesList);
    axios.get(`${BASE_SERVICE_URL}/activity?size=10&page=${page}&searchQuery=${this.props.searchQuery}`).then(response => {
      const { records, last } = response.data;
      this.setState({activitiesList: newActivities.concat(records), hasMoreActivities:!last});
    });
  }

  render() {
    const { hasMoreActivities, activitiesList } = this.state; 
      return (
        <div className="c-activity-list">
          <InfiniteScroll
            pageStart={0}
            loadMore={this.loadMoreActivities}
            hasMore={hasMoreActivities}
            loader={<LoadingAnimation className="s-activities-loading"/>}
            useWindow={true}
          >
            {
              activitiesList.map(activity => {
              return <ActivityCard activity={activity} history={this.props.history}/>
              })
            }
          </InfiniteScroll>
          <ScrollUp />
        </div>
      );
    }
}

export default ActivityList;
