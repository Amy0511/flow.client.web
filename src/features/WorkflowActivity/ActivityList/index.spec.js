import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import ActivityList from "./index";

const history = {};
const activities = [
  {
    creationDate: 1541070397344,
    duration: 13886,
    id: "5bdade3d37dc4700011b06c3",
    initiatedBy: "5bc7b126f7856000012cd95d",
    status: "completed",
    workflowId: "5bd9d8ab7eb44d0001772e64",
    workflowRevisionid: "5bd9e1cfa40f5d0001249bfa",
    description: "file ingestion",
    teamName: "Isa's Team",
    icon: "flow",
    shortDescription: "file ingestion",
    workflowName: "Business Process Test"
  }
];

const match = {
  url: "test.url"
};

const mockfn = jest.fn();

const props = {
  activities,
  history,
  match,
  loadMoreActivities: mockfn,
  setMoreActivities: mockfn
};

describe("ActivityList --- Snapshot", () => {
  it("Capturing Snapshot of ActivityList", () => {
    const renderedValue = renderer.create(
      <MemoryRouter>
        <ActivityList {...props} />
      </MemoryRouter>
    );
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("ActivityList --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <MemoryRouter>
        <ActivityList {...props} />
      </MemoryRouter>
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
