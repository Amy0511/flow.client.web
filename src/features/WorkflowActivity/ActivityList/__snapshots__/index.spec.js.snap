// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityList --- Snapshot Capturing Snapshot of ActivityList 1`] = `
Array [
  <div
    className="c-activity-list"
  >
    <a
      className="c-activities-card --completed"
      href="/activity/5bd9d8ab7eb44d0001772e64/execution/5bdade3d37dc4700011b06c3"
      onClick={[Function]}
    >
      <div
        className="c-activity-card-workflow"
      >
        <div
          className="c-activity-card-section--left"
        >
          <div
            className="c-activity-card__icon"
          >
            <img
              alt="Worklflow icon"
              className="b-activity-card__icon"
              src="flow.svg"
            />
          </div>
          <div
            className="c-activity-card-workflow-info"
          >
            <label
              className="b-activity-card__name"
            >
              Business Process Test
            </label>
            <label
              className="b-activity-card__description"
            >
              file ingestion
            </label>
          </div>
        </div>
        <div
          className="c-activity-card-section--right"
        >
          <ul
            className="c-activity-card-execution-info --first"
          >
            <li
              className="b-activity-card__row"
            >
              <label
                className="b-activity-card__label"
              >
                Team
              </label>
              <label
                className="b-activity-card__data"
              >
                Isa's Team
              </label>
            </li>
            <li
              className="b-activity-card__row"
            >
              <label
                className="b-activity-card__label"
              >
                Trigger
              </label>
              <label
                className="b-activity-card__data"
              >
                ---
              </label>
            </li>
            <li
              className="b-activity-card__row"
            >
              <label
                className="b-activity-card__label"
              >
                Initiated by
              </label>
              <label
                className="b-activity-card__data"
              >
                ---
              </label>
            </li>
          </ul>
          <ul
            className="c-activity-card-execution-info"
          >
            <li
              className="b-activity-card__row"
            >
              <label
                className="b-activity-card__label"
              >
                Start Time
              </label>
              <label
                className="b-activity-card__data"
              >
                2018-11-01 07:06 AM
              </label>
            </li>
            <li
              className="b-activity-card__row"
            >
              <label
                className="b-activity-card__label"
              >
                Duration
              </label>
              <label
                className="b-activity-card__data"
              >
                13 seconds
              </label>
            </li>
            <li
              className="b-activity-card__row"
            >
              <label
                className="b-activity-card__label"
              >
                 Status
              </label>
              <label
                className="b-activity-card__data"
              >
                <img
                  alt="Status completed"
                  className="b-activity-card__status-icon"
                  src="passed.svg"
                />
                Succeeded
              </label>
            </li>
          </ul>
        </div>
      </div>
    </a>
  </div>,
  <aside
    aria-label="Scroll to top of page"
    className="c-activities__scroll-up "
    data-testid="react-scroll-up-button"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={Object {}}
    tabIndex={-1}
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <img
        alt="chevron"
        className="b-activities__scroll-button"
        src="doublechevron.svg"
      />
      <label
        className="b-activities__scroll-label"
      >
        Go to top
      </label>
    </div>
  </aside>,
]
`;
